src/onnx/onnx.cpp

```diff
@@ -1807,6 +1808,39 @@ struct onnx_parser
         return ret_ins;
     }
 
+    instruction_ref
+    parse_onehot(const std::string&, node_info info, std::vector<instruction_ref> args)
+    {
+        migraphx::argument depth_arg = args[1]->eval();
+        check_arg_empty(depth_arg, "ONEHOT: depth - dynamic shape not supported");
+        size_t depth = depth_arg.at<size_t>();
+
+        int64_t axis = -1;
+        std::vector<float> on_off_vals;
+
+        migraphx::argument values_arg = args[2]->eval();
+        check_arg_empty(values_arg, "ONEHOT: values - dynamic shape not supported");
+        values_arg.visit([&](auto v) { copy(v, std::back_inserter(on_off_vals)); });
+        float off_value = on_off_vals[0];
+        float on_value  = on_off_vals[1];
+
+        std::vector<float> depth_input(depth * depth, off_value);
+        for(int i = 0; i < depth; i++)
+        {
+            depth_input[depth * i + i] = on_value;
+        }
+
+        if(contains(info.attributes, "axis"))
+            axis = info.attributes.at("axis").i();
+        if(axis == -1)
+        {
+            shape s{shape::float_type, {depth, depth}};
+            auto l0 = prog.add_literal({s, depth_input});
+            return prog.add_instruction(op::gather{0}, {l0, args[0]});
+        }
+        MIGRAPHX_THROW("MIGraphX does not support axis != -1");
```

Comment:
If you want, can you change the error message to "ONEHOT:  MIGraphX does not support axis != -1"?

Reply:
fixed